### PR TITLE
Export Class type in JS library

### PR DESCRIPTION
**Pull request by gw000**
The type `Class<T>` appears in the arguments to data filtering API functions, where it represents a constructor for type `T`: eg. in `oso.authorizedQuery(actor, action, Resource)` `Resource` has type `Class<T>`.

`Class` is defined in `types.ts`. This PR exports it from `index.ts` so users can access it more easily.

PR checklist:

- [x] Added changelog entry.

### CODE REVIEW

File: languages/js/src/index.ts [hunk 1]
```diff
@@ -3,3 +3,4 @@ export { Variable } from './Variable';
 export { AuthorizationError, ForbiddenError, NotFoundError } from './errors';
 export { Relation } from './dataFiltering';
 export { defaultEqualityFn } from './helpers';
+export { Class } from './types';
```
Comment (gj) [inline languages/js/src/index.ts:6]: Is this okay?
```suggestion
export type { Class } from './types';
```

File: docs/content/any/project/changelogs/NEXT.md [hunk 2]
```diff
@@ -16,6 +16,12 @@ draft: true
 
 - Fixed a bug that caused the Go library to crash when using Go 1.17 on macOS.
 
+### Node.js
+
+#### Other bugs & improvements
+- The `Class` type for representing abstract resources for data filtering is
```
Comment (gj) [inline docs/content/any/project/changelogs/NEXT.md:22]: This isn't really what the `Class` type is for, at least internally. Is this the only context in which someone would encounter the `Class` type while using the library?

File: languages/js/src/index.ts (hunk 1 above)
Comment (gj) [reply languages/js/src/index.ts:6]: Also, if we're making it a public export, we should add/update the doc string on the type: https://github.com/osohq/oso/blob/054b3ddae78a1f02e26fc74090a15ca95f4c493c/languages/js/src/types.ts#L316-L320

**Review by gj — CHANGES_REQUESTED** [review]
How did you test this change? Do you think we should add a note to the docs about when/how to use this newly exported type? Is this the only change required across the whole JS lib public API, or is this just the first fix and you're still evaluating the non-`authorizedResources()` APIs?

File: languages/js/src/index.ts (hunk 1 above)
Comment (gw000) [reply languages/js/src/index.ts:6]: I have no idea what the difference is between `export` and `export type`, can you explain

**Comment by gw000** [comment]
> How did you test this change? Do you think we should add a note to the docs about when/how to use this newly exported type? Is this the only change required across the whole JS lib public API, or is this just the first fix and you're still evaluating the non-`authorizedResources()` APIs?

I didn't test it, I'm just assuming JavaScript's module system works.

This is the only thing I've looked at. I'm still a little fuzzy on the motivation for this change honestly.

File: docs/content/any/project/changelogs/NEXT.md (hunk 2 above)
Comment (gw000) [reply docs/content/any/project/changelogs/NEXT.md:22]: It's the only example on record afaik ...

**Review by gj — APPROVED** [review]
:shipit:

**Pull request merged**